src/member/member.infra.ts

```diff
@@ -330,4 +343,102 @@ export class MemberInfrastructure {
 
     return queryBuilder;
   }
+
+  public async deleteMemberByEmail(appId: string, email: string, entityManager: EntityManager): Promise<DeleteResult> {
+    return entityManager.transaction(async (manager) => {
+      const memberRepo = manager.getRepository(Member);
+      const memberCategoryRepo = manager.getRepository(MemberCategory);
+      const memberTagRepo = manager.getRepository(MemberTag);
+      const memberOauthRepo = manager.getRepository(MemberOauth);
+      const memberDeviceRepo = manager.getRepository(MemberDevice);
+      const memberPhoneRepo = manager.getRepository(MemberPhone);
+      const memberPropertyRepo = manager.getRepository(MemberProperty);
+      const memberPermissionExtraRepo = manager.getRepository(MemberPermissionExtra);
+      const memberTrackingLogRepo = manager.getRepository(MemberTrackingLog);
+      const memberNoteRepo = manager.getRepository(MemberNote);
+      const memberTaskRepo = manager.getRepository(MemberTask);
+
+      const programContentProgressRepo = manager.getRepository(ProgramContentProgress);
+      const programContentLogRepo = manager.getRepository(ProgramContentLog);
+
+      const notificationRepo = manager.getRepository(Notification);
+      const couponRepo = manager.getRepository(Coupon);
+      const paymentLogRepo = manager.getRepository(PaymentLog);
+      const invoiceRepo = manager.getRepository(Invoice);
+      const orderProductRepo = manager.getRepository(OrderProduct);
+      const orderDiscountRepo = manager.getRepository(OrderDiscount);
+      const orderLogRepo = manager.getRepository(OrderLog);
+
+      const member = await memberRepo.findOneByOrFail([{ email: email, appId: appId }]);
+
+      const notifications = await notificationRepo.findBy([
```

Comment:
this find condition will be very slow 
use targetMemberId or sourceMemberId  to find 

or 

use delete instead of remove